### PR TITLE
virtual func return type check implemented

### LINKED ISSUE #811: _to_string should give an error if it don't return a string
**Describe the project you are working on:**
Making small programs to learn GDscript

**Describe the problem or limitation you are having in your project:**
I couldn't figure out why I couldn't change how my object was printed. I had forgotten to type return before the string.

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
If _to_string had a default return type I would have cough the error immediately.
 
**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
_to_string should give an error if it doesn't return a string. I assume it could use the mechanisms that are used to catch mixmatch returns with declared return types. 
**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
Not that I know of.
**Is there a reason why this should be core and not an add-on in the asset library?:**
It's most useful to new programmers that would have the experience to look for and install an addon.

### PR DESCRIPTION
**Pull request by ThakeeNathees**
Fix: https://github.com/godotengine/godot-proposals/issues/811

override-return-type

override-return-type-2

### CODE REVIEW

File: modules/gdscript/gdscript_parser.cpp [hunk 1]
```diff
@@ -7991,6 +7991,8 @@ void GDScriptParser::_check_function_types(FunctionNode *p_function) {
 		if (_get_function_signature(base_type, p_function->name, return_type, arg_types, default_arg_count, _static, vararg)) {
 			bool valid = _static == p_function->_static;
 			valid = valid && return_type == p_function->return_type;
+			if (!p_function->return_type.has_type)
+				p_function->return_type = return_type;
```
Comment (akien-mga) [inline modules/gdscript/gdscript_parser.cpp:7995]: This should now use braces around the single-line block.

Comment (ThakeeNathees) [reply modules/gdscript/gdscript_parser.cpp]: done!

**Comment by Calinou** [comment]
Now that the new GDScript has been merged in the `master` branch, we should check if this functionality has already been implemented in the `master` branch. If so, maybe there's still value in reopening this pull request against the `3.x` branch and merging it for 3.4.

**Comment by akien-mga** [comment]
I tested and the case is in the second screenshot this PR is properly implemented in `master`:
```
func _override_return_int() -> String:
	return "str"
```
will raise an error.

When not typing the override though, it doesn't raise an error.

I opened #73153 to track it.

This PR is abandoned so closing.

**Pull request closed without merging**